gitpod/gitpod.Dockerfile

```diff
@@ -1,48 +1,8 @@
-# Doing a local shallow clone - keeps the container secure
-# and much slimmer than using COPY directly or making a
-# remote clone
-ARG BASE_CONTAINER="pandas/pandas-dev:latest"
-FROM gitpod/workspace-base:latest as clone
+FROM condaforge/miniforge3:23.3.1-1
```

Comment:
One potential issue with using the conda image here is that it is pinned to a specific version. This can lead to Gitpod breaking like in #53685 when the version of conda becomes incompatible with the version of python and other dependencies the environment.yml file.

I think building from the base repo's dockerfile without conda is probably the simpler and easier to maintain approach, as the gitpod instance will update with the base repo's dockerfile. If we were to go down that route, the gitpod folder could actually be deleted entirely, and a .gitpod.yml file is all that is required.

Reply:
The reason I switched to conda from Docker is that docker doesn't provide a way to install packages (I don't think there is any sudo in the gitpod container - at least apt-get/pip don't work for me).

I think Docker might also have the same problem since the packages installed will be cached and won't update unless you recreate the gitpod instance.

I avoid the problem here by updating the conda packages on install based on the environment.yml. I'm happy to switch back to Docker if there's a way to do the same there.

Reply:
I _think_ most Docker images do not include sudo. What error were you seeing when doing something like `apt-get update && apt-get install -y <some_package>`?

Reply:
Something like this
```
gitpod@lithomas1-pandas-1tcur93d5vv:/workspace/pandas$ apt install build-essential
E: Could not open lock file /var/lib/dpkg/lock-frontend - open (13: Permission denied)
E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?
gitpod@lithomas1-pandas-1tcur93d5vv:/workspace/pandas$ sudo apt install build-essential
bash: sudo: command not found
```

Reply:
Hmm strange...and you are running that between the

```sh
USER root

....

USER gitpod
```

commands in the image right? I think anything in between those should be running under the root user account

Reply:
No, this is when the workspace is started up.

Reply:
It looks like the Dockerfile intentionally runs under a non-priviledged user. So you'd have to install system libraries before that or remove the `USER gitpod` statement in `gitpod.Dockerfile` (no idea if that is a good idea for gitpod or not)